### PR TITLE
Feature/update database

### PR DESCRIPTION
**Pull request by GagarinPavel**

### CODE REVIEW

File: AIS/AIS.API/ViewModels/QuestionAreasQuestionSets/QuestionAreasQuestionSets.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿
+using AIS.API.ViewModels.QuestionArea;
+using AIS.API.ViewModels.QuestionSet;
+
+namespace AIS.API.ViewModels.QuestionAreasQuestionSets
+{
+    public class QuestionAreasQuestionSets
```
Comment (nantonov) [inline AIS/AIS.API/ViewModels/QuestionAreasQuestionSets/QuestionAreasQuestionSets.cs:7]: Add ViewModel postfix please

File: AIS/AIS.API/ViewModels/QuestionAreasQuestionSets/QuestionAreasQuestionSets.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    {
+        public int Id { get; set; }
+        public int QuestionAreaId { get; set; }
+        public ShortQuestionAreaViewModel QuestionArea { get; set; }
+        public int QuestionSetId { get; set; }
+        public virtual ShortQuestionSetViewModel QuestionSet { get; set; }
```
Comment (nantonov) [inline AIS/AIS.API/ViewModels/QuestionAreasQuestionSets/QuestionAreasQuestionSets.cs:13]: not needed virtual in vm

File: AIS/AIS.API/ViewModels/QuestionsQuestionSets/QuestionsQuestionSetsViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+using AIS.API.ViewModels.QuestionSet;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AIS.API.ViewModels.QuestionsQuestionSets
+{
+    public class QuestionsQuestionSets
```
Comment (nantonov) [inline AIS/AIS.API/ViewModels/QuestionsQuestionSets/QuestionsQuestionSetsViewModel.cs:11]: Add ViewModel postfix please

File: AIS/AIS.API/ViewModels/QuestionsQuestionSets/QuestionsQuestionSetsViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    {
+        public int Id { get; set; }
+        public int QuestionId { get; set; }
+        public virtual ShortQuestionViewModel Question { get; set; }
+        public int QuestionSetId { get; set; }
+        public virtual ShortQuestionSetViewModel QuestionSet { get; set; }
+}
```
Comment (nantonov) [inline AIS/AIS.API/ViewModels/QuestionsQuestionSets/QuestionsQuestionSetsViewModel.cs:18]: format the file please

File: AIS/AIS.BLL/Models/Question.cs [hunk 5]
```diff
@@ -1,11 +1,12 @@
...
+
+namespace AIS.BLL.Models
 {
     public class Question
     {
         public int Id { get; set; }
         public string Text { get; set; }
-        public int QuestionSetId { get; set; }
-        public QuestionSet QuestionSet { get; set; }
+        public virtual IEnumerable<QuestionSet> QuestionSets { get; set; }
```
Comment (nantonov) [inline AIS/AIS.BLL/Models/Question.cs:9]: No needed virtual here

File: AIS/AIS.BLL/Models/QuestionsQuestionSets.cs [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+namespace AIS.BLL.Models
+{
+    public class QuestionsQuestionSets
+    {
+        public int Id { get; set; }
+        public int QuestionId { get; set; }
+        public virtual Question Question { get; set; }
+        public int QuestionSetId { get; set; }
+        public virtual QuestionSet QuestionSet { get; set; }
+}
```
Comment (nantonov) [inline AIS/AIS.BLL/Models/QuestionsQuestionSets.cs:16]: format please

File: AIS/AIS.BLL/Models/QuestionsQuestionSets.cs [hunk 7]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AIS.BLL.Models
+{
+    public class QuestionsQuestionSets
```
Comment (nantonov) [inline AIS/AIS.BLL/Models/QuestionsQuestionSets.cs:9]: No needed virtuals in this file

File: AIS/AIS.BLL/Services/NextQuestionService.cs [hunk 8]
```diff
@@ -36,7 +36,7 @@ public async Task<Question> NextQuestion(int sessionId, CancellationToken ct)
             {
                 foreach(var set in sets)
                 {
-                    if (answers.Count(answer => answer.Question.QuestionSetId == set.Id) != set.Questions.Count)
+                    if (answers.Count(answer => answer.Question.QuestionSets.Any(s=> s.Id == set.Id)) != set.Questions.Count)
```
Comment (nantonov) [inline AIS/AIS.BLL/Services/NextQuestionService.cs:39]: What happens here?

**Review by nantonov — CHANGES_REQUESTED** [review]

**Review by nantonov — APPROVED** [review]

**Review by Blackolf931 — APPROVED** [review]

**Review by stas1408123 — APPROVED** [review]

**Pull request merged**